lib/compiler-singlepass/src/machine_arm64.rs

```diff
@@ -5185,46 +5174,37 @@ impl Machine for MachineARM64 {
     }
     fn i64_popcnt(&mut self, loc: Location, ret: Location) -> Result<(), CompileError> {
         let mut temps = vec![];
-        let src = self.location_to_reg(Size::S64, loc, &mut temps, ImmType::None, true, None)?;
-        let dest = self.location_to_reg(Size::S64, ret, &mut temps, ImmType::None, false, None)?;
-        let src = if src == loc {
-            let tmp = self.acquire_temp_gpr().ok_or_else(|| {
-                CompileError::Codegen("singlepass cannot acquire temp gpr".to_owned())
-            })?;
-            temps.push(tmp);
-            self.assembler
-                .emit_mov(Size::S64, src, Location::GPR(tmp))?;
-            Location::GPR(tmp)
-        } else {
-            src
-        };
-        let tmp = {
-            let tmp = self.acquire_temp_gpr().ok_or_else(|| {
-                CompileError::Codegen("singlepass cannot acquire temp gpr".to_owned())
-            })?;
-            temps.push(tmp);
-            Location::GPR(tmp)
-        };
-        let label_loop = self.assembler.get_label();
-        let label_exit = self.assembler.get_label();
-        self.assembler
-            .emit_mov(Size::S32, Location::GPR(GPR::XzrSp), dest)?; // dest <= 0
-        self.assembler.emit_cbz_label(Size::S64, src, label_exit)?; // src == 0, then goto label_exit
-        self.assembler.emit_label(label_loop)?;
```

Comment:
We need to use this codepath if the chipset doesn't support NEON (fallback).
So please wrap the old / new code in a if/else